src/S3/StreamWrapper.php

```diff
@@ -592,16 +605,24 @@ private function validate($path, $mode)
                 . "Use one 'r', 'w', 'a', or 'x'.";
         }
 
-        // When using mode "x" validate if the file exists before attempting
-        // to read
-        if ($mode == 'x' &&
-            $this->getClient()->doesObjectExist(
-                $this->getOption('Bucket'),
-                $this->getOption('Key'),
-                $this->getOptions(true)
-            )
-        ) {
-            $errors[] = "{$path} already exists on Amazon S3";
+        if ($mode === 'x') {
+            if (self::$useV2Existence) {
```

Comment:
Same as above, there's a way to reformat this.  You can also have a string as the variable method, and have it be 
```
$method = self::$useV2Existence ? "doesObjectExistV2" : "doesObjectExist";
$this->getClient()->$method();
```

Look at example 2 [here](https://www.php.net/manual/en/functions.variable-functions.php).

There is a precedent for this being used in our code, and if you choose this way, you can make the checkBucketExistance method I mentioned above into just checkExistance, which takes in $type (bucket or object) and $name (for instance, bucket name).

Reply:
I like this idea better.  Much cleaner than I imagined it